napari/layers/labels/labels.py

```diff
@@ -465,9 +506,20 @@ def _raw_to_displayed(self, raw):
         image : array
             Image mapped between 0 and 1 to be displayed.
         """
-        image = np.where(
-            raw > 0, colormaps._low_discrepancy_image(raw, self._seed), 0
-        )
+        if self.enable_custom_color_dict:
+            u, inv = np.unique(raw, return_inverse=True)
+            image = np.array(
+                [
+                    self.custom_color[x]
+                    if x in self.custom_color
+                    else colormaps._low_discrepancy_image(x, self._seed)
```

Comment:
```suggestion
                    else 1.0
```
I'd make this change and then adjust the colormap so that 1.0 always maps to black (or potentially some other color), rather than reuse the random color map here too